hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/processors/CustomContractCreationProcessor.java

```diff
@@ -16,20 +16,81 @@
+import static com.hedera.node.app.service.contract.impl.exec.utils.FrameUtils.proxyUpdaterFor;
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater;
 import edu.umd.cs.findbugs.annotations.NonNull;
 import java.util.List;
+import java.util.Optional;
+import org.hyperledger.besu.datatypes.Address;
 import org.hyperledger.besu.evm.EVM;
+import org.hyperledger.besu.evm.account.MutableAccount;
 import org.hyperledger.besu.evm.contractvalidation.ContractValidationRule;
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.hyperledger.besu.evm.frame.MessageFrame;
 import org.hyperledger.besu.evm.gascalculator.GasCalculator;
 import org.hyperledger.besu.evm.processor.ContractCreationProcessor;
+import org.hyperledger.besu.evm.tracing.OperationTracer;
 
+/**
+ * A customization of the Besu {@link ContractCreationProcessor} that replaces the
+ * explicit {@code sender.decrementBalance(frame.getValue())} and
+ * {@code contract.incrementBalance(frame.getValue())} calls with a single call
+ * to the {@link HederaWorldUpdater#tryTransferFromContract(Address, Address, long, boolean)}
+ * dispatch method.
+ */
 public class CustomContractCreationProcessor extends ContractCreationProcessor {
+    // By convention, the halt reason should be INSUFFICIENT_GAS when the contract already exists
+    private static final Optional<ExceptionalHaltReason> COLLISION_HALT_REASON =
+            Optional.of(ExceptionalHaltReason.INSUFFICIENT_GAS);
+
     public CustomContractCreationProcessor(
             @NonNull final EVM evm,
             @NonNull final GasCalculator gasCalculator,
             final boolean requireCodeDepositToSucceed,
             @NonNull final List<ContractValidationRule> contractValidationRules,
             final long initialContractNonce) {
-        super(gasCalculator, evm, requireCodeDepositToSucceed, contractValidationRules, initialContractNonce);
+        super(
+                requireNonNull(gasCalculator),
+                requireNonNull(evm),
+                requireCodeDepositToSucceed,
+                requireNonNull(contractValidationRules),
+                initialContractNonce);
+    }
+
+    @Override
+    public void start(@NonNull final MessageFrame frame, @NonNull final OperationTracer tracer) {
+        final var addressToCreate = frame.getContractAddress();
+        final var contract =
+                frame.getWorldUpdater().getOrCreate(addressToCreate).getMutable();
+
+        if (alreadyCreated(contract)) {
+            halt(frame, COLLISION_HALT_REASON);
+            tracer.traceAccountCreationResult(frame, COLLISION_HALT_REASON);
+        } else {
+            final var updater = proxyUpdaterFor(frame);
+            // A contract creation is never a delegate call, hence the false argument below
+            final var maybeReasonToHalt = updater.tryTransferFromContract(
```

Comment:
Q: just checking does the no value scenario get covered here? That is does it work and not fail when no value was provided to transfer. Something to add to tests if not already covered.

Reply:
I'm thinking this method should just immediately return `Optional.empty()` if asked to transfer zero value.

Will add that to its javadoc in my next PR.